include/mxnet/ndarray.h

```diff
@@ -456,13 +456,15 @@ class NDArray {
   // Wrap the tblob of aux data into an NDArray which shares the same variable with the
   // current one.
   inline const NDArray aux_ndarray(size_t i) const {
+    this->WaitToRead();
```

Comment:
I don't have wait_to_read/write for `TBlob aux_data()` because I assume accessing these are protected by the engine for thread safety. But aux_ndarray can be called when frontend call x.indices/x.indptr. Instead of letting user call x.wait_to_read explicitly, I moved this wait to backend